platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/StateSigningTests.java

```diff
@@ -81,44 +85,40 @@ void addValidSignaturesTest(final boolean evenWeighting) {
 
         final int nodeCount = random.nextInt(10, 20);
 
-        final AddressBook addressBook = RandomAddressBookBuilder.create(random)
+        final Roster roster = RandomRosterBuilder.create(random)
                 .withWeightDistributionStrategy(
-                        evenWeighting
-                                ? RandomAddressBookBuilder.WeightDistributionStrategy.BALANCED
-                                : RandomAddressBookBuilder.WeightDistributionStrategy.GAUSSIAN)
+                        evenWeighting ? WeightDistributionStrategy.BALANCED : WeightDistributionStrategy.GAUSSIAN)
                 .withSize(nodeCount)
                 .build();
-
         final SignedState signedState = new RandomSignedStateGenerator(random)
-                .setAddressBook(addressBook)
+                .setAddressBook(RosterUtils.buildAddressBook(roster))
```

Comment:
Minor: since the SignedState is now converted to use Rosters, do we want to update its generator to also only ever use rosters? This can be done in a separate PR if you like, in which case please file a new ticket to track this work.

Reply:
I'll create a new ticket. Updating `RandomSignedStateGenerator` was sorta blocked by your work and will become unblocked once your [PR](https://github.com/hashgraph/hedera-services/pull/17047) is merged in.

Reply:
Created https://github.com/hashgraph/hedera-services/issues/17087